microsoft-edge/dualengine/concepts/adapter-dll.md

```diff
@@ -0,0 +1,79 @@
+With that said the DualEngine API is not a typical LAF and 
+`Windows.ApplicationModel.TryUnlockFeature` is not used to unlock the feature. This is
+because LAF typically uses the application identity of the calling process to grant access, as a plugin DLL this
+will always be Internet Explorer. Therefore to unlock the API you are required to call 
+[DualEngineSessionFactory::TryUnlockFeature](../reference/dualenginesessionfactory.md#tryunlockfeature).
+
+In addition the identity of you're application is required to be provided to Internet Explorer via a specific
+resource string set in your DLL, you can set the value in your resource file like so:
+> IDENTITY LIMITEDACCESSFEATURE { L"ExampleApp_6v1kqc8g0gcae" }
+
+Your specific identity string will be provided to you by Microsoft when LAF access is granted.
+
+## Exports
+It is required that your adapter DLL implement a number of exports. These exports are called by Internet Explorer
+when loading your DLL and provide you with the factory objects for creating DualEngine API objects.
+
+It should be noted that these exports are called from the main thread of Internet Explorer process so doing any
+long term processing inside the exports will halt Internet Explorer, as such it is recommended to create a new
+thread to handle any long running work.
+
+#### DualEngineInitialize
+> HRESULT APIENTRY DualEngineInitialize(DualEngineSessionFactory* pSessionFactory, PCWSTR pipeName)
+
+##### Parameters
+* `pSessionFactory` The factory object to use to access the DualEngine API
+* `pipeName` A string that was passed in to Internet Explorer via the `-DualEnginePipe` command line flag.
+
+This export is called immediately after Internet Explorer successfully loads the DLL. This is your first chance
+to set up anything necessary for your application.
+
+Presumably this is where you would setup communication between the adapter DLL and your host App, the `pipeName`
+string is provided for this purpose. Despite it's name this need not be a pipe name, it is simply a string passed 
+to your adapter based on the value of `-DualEnginePipe` command line flag Internet Explorer was launched with and
+has no other semantic meaning.
+
+Also note that while the `pSessionFactory` object is provided in this call it is currently not valid to call 
+`GetVersionedBrowserSession` here as Internet Explorer is currently not ready to create Sessions.
+
+
+#### DualEngineReady
+> HRESULT APIENTRY DualEngineReady()
+
+This export is called after Internet Explorer has finished all of it's initial setup and the API is ready to use. 
+It is now possible to call `GetVersionedBrowserSession` and get the Session object assuming a successful call to 
+`TryUnlockFeature` has been made.
+
+## Signing
+Internet Explorer has a requirement that an adapter DLL must be signed by a trusted signature for it to be loaded.
+
+For testing and development purposes this check can be bypassed by turning TestSigning on for the device you're testing on.

```

Comment:
```suggestion
For testing and development purposes, this check can be bypassed by turning TestSigning on for the device you're testing on.
```

Reply:
Internet Explorer has a requirement that for an adapter DLL to be loaded, it must be signed with a trusted signature.  For testing and development purposes, this check can be bypassed, by turning on `TestSigning` for the device that you're testing on, as follows: